integration-tests/integration_test.go

```diff
@@ -0,0 +1,931 @@
+		issuerRequest.MaxTokens,
+		now.Format(time.RFC3339),
+		expires.Format(time.RFC3339),
+		issuerRequest.Duration,
+		issuerRequest.Overlap,
+		issuerRequest.Buffer,
+	)
+	createTestIssuer(t, issuerRequest)
+
+	t.Run("issue_tokens_via_kafka", func(t *testing.T) {
+		tokens, blindedTokens, signingResultSet = issueTokensViaKafka(
+			t,
+			testMetadata,
+			testID,
+			requestID,
+			issuerRequest,
+		)
+		t.Logf(
+			"Successfully received signing results with %d tokens and %d result entries",
+			len(tokens),
+			len(signingResultSet.Data),
+		)
+		require.NotEmpty(t, tokens, "Should have tokens after issuance")
+		require.NotEmpty(t, blindedTokens, "Should have blinded tokens after issuance")
+		require.NotEmpty(t, signingResultSet.Data, "Should have signing results after issuance")
+	})
+
+	t.Run("map_tokens_to_signing_keys", func(t *testing.T) {
+		for i, originalBlindedToken := range blindedTokens {
+			t.Logf("Processing token %d/%d...", i+1, len(blindedTokens))
+			originalBlindedTokenBytes, _ := originalBlindedToken.MarshalText()
+			originalBlindedTokenStr := string(originalBlindedTokenBytes)
+			var foundMatch bool
+			for j, result := range signingResultSet.Data {
+				for k, resultBlindedTokenStr := range result.Blinded_tokens {
+					if resultBlindedTokenStr == originalBlindedTokenStr {
+						t.Logf(
+							"Found matching blinded token in result %d at position %d",
+							j,
+							k,
+						)
+						foundMatch = true
+
+						var batchDLEQProof crypto.BatchDLEQProof
+						_ = batchDLEQProof.UnmarshalText([]byte(result.Proof))
+						var issuerPublicKey crypto.PublicKey
+						_ = issuerPublicKey.UnmarshalText([]byte(result.Issuer_public_key))
+						var signedToken crypto.SignedToken
+						_ = signedToken.UnmarshalText([]byte(result.Signed_tokens[k]))
+						var blindedToken crypto.BlindedToken
+						_ = blindedToken.UnmarshalText([]byte(resultBlindedTokenStr))
+
+						resultUnblindedTokens, err := batchDLEQProof.VerifyAndUnblind(
+							[]*crypto.Token{tokens[i]},
+							[]*crypto.BlindedToken{&blindedToken},
+							[]*crypto.SignedToken{&signedToken},
+							&issuerPublicKey)
+
+						require.NoError(
+							t,
+							err,
+							"Should successfully verify and unblind",
+						)
+						require.Len(
+							t,
+							resultUnblindedTokens,
+							1,
+							"Should get exactly one unblinded token",
+						)
+
+						allTokenInfos = append(allTokenInfos, tokenInfo{
+							UnblindedToken: resultUnblindedTokens[0],
+							SignedKey:      result.Issuer_public_key,
+						})
+						break
+					}
+				}
+				if foundMatch {
+					break
+				}
+			}
+			require.True(
+				t,
+				foundMatch,
+				"token %d should have a matching result",
+				i+1,
+			)
+		}
+		require.Len(t, allTokenInfos, len(tokens),
+			"Should have the same number of token infos as original tokens")
+	})
+
+	for i, token := range allTokenInfos {
+		if i%2 == 0 {
```

Comment:
Its not clear why `i%2 == 0` would result in calling different endpoints i.e. `v1` or `v3`. Is this just to test both endpoints with half the tokens?

Reply:
Yes it's just to distribute the tokens across both endpoints.